tests/TestCase/ResultSetTest.php

```diff
@@ -92,13 +99,78 @@ public function testCurrent($resultSets)
         $this->assertEquals('things', $document->type());
     }
 
+    /**
+     * Tests that calling current will get converted fields
+     * basic types
+     *
+     * @return void
+     */
+    public function testBasicTypesConversions()
+    {
+        $connection = ConnectionManager::get('test');
+
+        $elasticaSet = $this->getMockBuilder('Elastica\ResultSet')
+            ->disableOriginalConstructor()
+            ->getMock();
+        $index = $this->getMockBuilder('Cake\ElasticSearch\Index')->getMock();
+        $query = $this->getMockBuilder('Cake\ElasticSearch\Query')
+            ->setConstructorArgs([$index])
+            ->getMock();
+        $query->expects($this->once())->method('getRepository')
+            ->will($this->returnValue($index));
+        $index->expects($this->once())
+            ->method('getEntityClass')
+            ->will($this->returnValue(__NAMESPACE__ . '\MyTestDocument'));
+        $index->method('embedded')
+            ->will($this->returnValue([]));
+        $index->expects($this->once())
+            ->method('getConnection')
+            ->will($this->returnValue($connection));
```

Comment:
Could we do a test with fewer mocks? Perhaps an integration test that fetches a document with a variety of types and checks the casted results in the document?

With this many mock is hard to be confident that we're testing actual code here and not just our mocks.

Reply:
Totally agree.